runtime/src/main/java/org/corfudb/runtime/ViewsMaintenance.java

```diff
@@ -13,49 +14,90 @@
 
 
 /**
- * A CorfuRuntime garbage collector. The runtime contains some views that track
+ * Periodic tasks to maintain the CorfuRuntime Views. The runtime contains some views that track
  * the global log, as sections of the global log get checkpointed and trimmed
  * the runtime's views need to remove log metadata that corresponds to the trimmed
- * parts of the log.
+ * parts of the log. Also, auto sync objects to amortize the log sync costs.
  * <p>
  * Created by Maithem on 11/16/18.
  */
 
 @Slf4j
 @NotThreadSafe
-public class ViewsGarbageCollector {
+public class ViewsMaintenance {
 
 
     private long trimMark = Address.NON_ADDRESS;
 
     @Getter
     private boolean started;
 
-    final ScheduledExecutorService gcThread = Executors.newSingleThreadScheduledExecutor(
+    /**
+     * How often to sync opened objects
+     */
+    @Getter
+    @Setter
+    long objectSyncDelayInSeconds = 30;
+
+    /**
+     * How often to run GC on the opened objects
+     */
+    @Getter
+    @Setter
+    long objectGCPeriodInSeconds = 20 * 60;
+
+    /**
+     * This single thread is shared by the two tasks (i.e. object GC and object syncing)
+     */
+    final ScheduledExecutorService maintenance = Executors.newSingleThreadScheduledExecutor(
             new ThreadFactoryBuilder().setDaemon(true)
-                    .setNameFormat("ViewsGarbageCollector")
+                    .setNameFormat("ViewsMaintenance")
                     .build());
 
     final CorfuRuntime runtime;
 
-    public ViewsGarbageCollector(CorfuRuntime runtime) {
+    public ViewsMaintenance(CorfuRuntime runtime) {
         this.runtime = runtime;
         this.started = false;
     }
 
     public void start() {
-        gcThread.scheduleAtFixedRate(() -> runRuntimeGC(),
-                runtime.getParameters().getRuntimeGCPeriod().toMinutes(),
-                runtime.getParameters().getRuntimeGCPeriod().toMinutes(),
-                TimeUnit.MINUTES);
+        maintenance.scheduleAtFixedRate(() -> runRuntimeGC(),
+                objectGCPeriodInSeconds, objectGCPeriodInSeconds,
+                TimeUnit.SECONDS);
+        maintenance.scheduleAtFixedRate(() -> runSyncObjects(),
+                objectSyncDelayInSeconds, objectGCPeriodInSeconds,
```

Comment:
is this right? initial delay = 30sec, and then run every 20min? or is it intended to run every 30secs? with such a big gap wouldn't it be less risk to separate both tasks?